tests/common-tests.h

```diff
@@ -288,8 +348,43 @@ template<> struct type_conversion<PhonebookEntry3>
     }
 };
 
+template<typename T>
+struct type_conversion<DdlTester<T>>
+{
+    typedef soci::values base_type;
+    static void from_base(soci::values const &v, soci::indicator, DdlTester<T>& t)
+    {
+        map_ddl_tester(t, v);
+    }
+    static void to_base(DdlTester<T> const &t, soci::values &v, soci::indicator&)
+    {
+        v.set("VAL", t.inVal);
+    }
+};
 } // namespace soci
 
+template<typename T>
+void test_limit(soci::session &sql, DdlTester<T>&& tester)
+{
+    const std::string table = "TEST_LIMIT";
+    try
+    {
+        sql.create_table(table).column("VAL", tester.inType);
+        sql << "INSERT INTO " << table << "(VAL) VALUES (:VAL)", soci::use(tester.inVal);
+        soci::statement stmt = (sql.prepare << "SELECT * FROM " << table);
+        stmt.exchange(soci::into(tester));
+        stmt.define_and_bind();
+        stmt.execute();
+        stmt.fetch();
+        check(tester);
+    }
+    catch (const std::exception& e)
+    {
+        CHECK_FALSE(e.what());
```

Comment:
```suggestion
        FAIL(e.what());
```

Reply:
Actually, `FAIL` is forcing a failure for the test case while `CHECK_FALSE` will still allow the test to continue, in this case, it makes more sense to keep `CHECK_FALSE`.

Reply:
Sorry, should have been `FAIL_CHECK()`.

But using `FAIL` is important, it makes _sense_, unlike `CHECK_FALSE(e.what())` which does not -- we're not checking for `e.what() == nullptr` at all here.

Reply:
II was looking at the macro in catch2/assertions.md instead of catch2/logging.md that's why I had trouble finding the macro.